tests/suites/test_suite_ecp.function

```diff
@@ -1038,23 +1086,138 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void ecp_set_public_key_group_check(int grp_id, int expected_ret)
+{
+    mbedtls_ecp_keypair key;
+    mbedtls_ecp_keypair_init(&key);
+    mbedtls_ecp_point Q;
+    mbedtls_ecp_point_init(&Q);
+
+    TEST_EQUAL(mbedtls_ecp_set_public_key(grp_id, &key, &Q),
+               expected_ret);
+
+exit:
+    mbedtls_ecp_keypair_free(&key);
+    mbedtls_ecp_point_free(&Q);
+}
+/* END_CASE */
+
+/* BEGIN_CASE */
+void ecp_set_public_key_good(int grp_id, data_t *public_data)
+{
+    mbedtls_ecp_keypair key;
+    mbedtls_ecp_keypair_init(&key);
+    mbedtls_ecp_group grp;
+    mbedtls_ecp_group_init(&grp);
+    mbedtls_ecp_point Q;
+    mbedtls_ecp_point_init(&Q);
+
+    TEST_EQUAL(mbedtls_ecp_group_load(&grp, grp_id), 0);
+    TEST_EQUAL(mbedtls_ecp_point_read_binary(&grp, &Q,
+                                             public_data->x, public_data->len),
+               0);
+
+    /* Freshly initialized key */
+    TEST_EQUAL(mbedtls_ecp_set_public_key(grp_id, &key, &Q), 0);
+    TEST_EQUAL(key.grp.id, grp_id);
+    TEST_EQUAL(mbedtls_ecp_point_cmp(&key.Q, &Q), 0);
+
+#if defined(MBEDTLS_BIGNUM_C)
```

Comment:
Hmm, it seems to me this entire test file is only compiled when `MBEDTLS_ECP_LIGHT` is enabled, and then that means so is `BIGNUM_C`. Am I missing something? What did you have in mind?

(If you're thinking about driver-only builds, those that have `BIGNUM_C` disabled simply don't compile the functions in this file.)

(More occurrences below, I won't point each of them.)

Reply:
I don't remember why I put guards in this function. To be robust with respect to what ECP light might have become, maybe? (I wrote this code back in June 2023.) Or to work with `MBEDTLS_ECP_ALT`? Anyway, the guards aren't bad, but they are noise unless we generalize the test suite, which, six months later, isn't going to happen. Should I remove those guards?

Reply:
`ECP_ALT` doesn't change anything here: the only parts of `ECP_C` it affects are outside of `ECP_LIGHT`, which does require `BIGNUM_C` and will continue doing so for the foreseeable future.

So IMO these guards are noise that should be removed, to avoid giving readers a false impression.